pkg/types/validation/installconfig.go

```diff
@@ -699,3 +703,35 @@ func validateFIPSconfig(c *types.InstallConfig) field.ErrorList {
 	}
 	return allErrs
 }
+
+// validateCapabilities checks if additional, optional openshift components are specified in the
+// install-config to be included in the installation
+func validateCapabilities(c *types.Capabilities, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if c.BaselineCapabilitySet == "" {
+		allErrs = append(allErrs, field.Required(fldPath.Child("baselineCapabilitySet"), "baseline capability set required"))
+
+		return allErrs
+	}
+	// Check to see if the additionalEnabledCapabilities is defined in openshift/api
+	if availableCapabilities, found := configv1.ClusterVersionCapabilitySets[c.BaselineCapabilitySet]; found {
+		// create a set of the available capability set
+		availableCapabilitySet := sets.NewString()
+		for _, capability := range availableCapabilities {
+			availableCapabilitySet.Insert(string(capability))
+		}
+		// create a set of the user specified additional capability set
+		additionalCapabilitySet := sets.NewString()
+		for _, capability := range c.AdditionalEnabledCapabilities {
+			additionalCapabilitySet.Insert(string(capability))
+		}
+		// Check for unknown capabilities between available and additional sets
+		missingCapability := additionalCapabilitySet.Difference(availableCapabilitySet)
+		for _, missing := range missingCapability.List() {
+			allErrs = append(allErrs, field.Invalid(fldPath, missing, fmt.Sprintf("unknown additional enabled capability specified")))
+		}
+	} else {
+		allErrs = append(allErrs, field.Invalid(fldPath, c.BaselineCapabilitySet, fmt.Sprintf("unknown baseline capability set specified")))
```

Comment:
possibly helpful to include a list of `configv1.ClusterVersionCapabilitySets` keys here, so folks see the valid choices and can copy/paste typo corrections?

Reply:
Done.